RecoVertex/PrimaryVertexProducer/interface/DAClusterizerInZT_vect.h

```diff
@@ -0,0 +1,318 @@
+#ifndef DAClusterizerInZT_vect_h
+#define DAClusterizerInZT_vect_h
+
+/**\class DAClusterizerInZT_vect
+
+ Description: separates event tracks into clusters along the beam line
+
+	Version which auto-vectorizes with gcc 4.6 or newer
+
+ */
+
+#include "RecoVertex/PrimaryVertexProducer/interface/TrackClusterizerInZ.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrack.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include <vector>
+#include "DataFormats/Math/interface/Error.h"
+#include "RecoVertex/VertexTools/interface/VertexDistanceXY.h"
+#include "RecoVertex/VertexPrimitives/interface/TransientVertex.h"
+
+#include <memory>
+#include <stdlib.h>
+#include <malloc.h>
+
+template <typename T, std::size_t N = 16>
+class AlignmentAllocator {
```

Comment:
@lgray 
please provide some evidence that the explicit align specifications provide significant gains in performance.
the implementation here uses simple data "regular" in byte sizes.
perhaps a simple ```#pragma simd``` in front of a loop is enough in most cases if the compiler is not convinced otherwise.

None of this magic is in RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc. If it actually helps, we should consider to propagate it there as well (although merging the two algos still seems more beneficial because the underalying logic is the same except for a few configurables).